.github/workflows/test.yml

```diff
@@ -169,3 +180,20 @@ jobs:
     - run: npm ls --all || true
     # Run a subset of tests that we can reasonably get running on Windows CI.
     - run: node test/test.js
+
+  # single status check that can be set as required in branch protection rules.
+  # success if all jobs listed in "needs" are successful.
+  # failure if at least one job listed "needs" is not successful.
+  test:
+    if: always()
+    runs-on: ubuntu-latest
+    needs:
+      - lint
+      - test-vers
+      - test-windows
+    steps:
+      - id: check
+        uses: elastic/apm-pipeline-library/.github/actions/check-dependent-jobs@current
+        with:
+          needs: ${{ toJSON(needs) }}
+      - run: ${{ steps.check.outputs.isSuccess }}
```

Comment:
Yes, the whole workflow will fail. Still, you won't have a single status check in the PR that can be set as required check.

This is usually not a big deal. However, in this case we have N status checks because of N matrix items. And there is no single status check for the whole workflow.

This job creates a status check which can be set as required.

Reply:
Ah, okay, thanks.

So the "check" names used for setting "required checks": are these the workflow names (e.g. `name: test` at the top of the file) or the **job** (or job matrix) name -- i.e. the `jobs:\n... test:` item)?

Reply:
Actually, it's only the job name.

The status check looks like this:

`<workflow name> / <job name>`

However, to make it required you only add the job name in the branch protection rules.

This means, actually any job with the same name in any workflow will be required.